.github/workflows/rspec_tests.yaml

```diff
@@ -0,0 +1,32 @@
+---
+name: RSpec test
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  rspec_tests:
+    name: ${{ matrix.cfg.os }}(ruby ${{ matrix.cfg.ruby }})
+    strategy:
+      matrix:
+        cfg:
+          - {os: ubuntu-18.04, ruby: '2.5'}
+          - {os: ubuntu-18.04, ruby: '2.7'}
+          - {os: windows-2019, ruby: '2.5'}
+          - {os: windows-2019, ruby: '2.7'}
+
+    runs-on: ${{ matrix.cfg.os }}
+    steps:
+      - name: Checkout current PR
+        uses: actions/checkout@v2
+
+      - name: Install ruby version ${{ matrix.cfg.ruby }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.cfg.ruby }}
+      
+      - run: "bundle exec rake spec"
+      - run: "bundle exec rake commits"
```

Comment:
I think this is going to run the `rake commits` tasks for each ruby/os combination. One option is to split the check into a separate file like we do in facter and puppet:

https://github.com/puppetlabs/puppet/blob/main/.github/workflows/checks.yaml